File: admin/src/models/customer.js
import { ModelBase } from './base'

export class Customer extends ModelBase {
    id = 0;
    customerName = "";
    name = "";
    emailAddress = "";
    addressLine1 = "";
    addressLine2 = "";
    city = "";
    provinceOrState = "";
    postalCode = 0;
    dateTime = new Date();

    constructor(dataObj) {
        super();        //By calling the super() method in the constructor method, we call the parent's constructor method and get access to the parent's properties and methods:
        this.mapData(dataObj);
    }
}

File: server/Models.System/Customer.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Models.System
{
    [Table("Customer")]
    public class Customer
    {
        [Key]
        public int ID { get; set; }

        [Required]
        [MaxLength(150)]
        public string CustomerName { get; set; }

        [Required]
        [MaxLength(500)]
        public string EmailAddress { get; set; }

        [Required]
        [MaxLength(250)]
        public string AddressLine1 { get; set; }

        [Required]
        [MaxLength(250)]
        public string AddressLine2 { get; set; }


        [Required]
        [MaxLength(100)]
        public string City { get; set; }


        [Required]
        [MaxLength(100)]
        public string ProvinceOrState { get; set; }


        [Required]
        [MaxLength(15)]
        public int PostalCode { get; set; }

    }
}
